frontend/src/libs/components/image-selector/image-selector.tsx

```diff
@@ -0,0 +1,182 @@
+  modalImageScale = Default.INITIAL_ZOOM,
+  viewMode = Default.VIEW_MODE,
+  className,
+  resultOptions,
+}: ImageSelectorProperties<T, K>): JSX.Element => {
+  const [isSelecting, setIsSelecting] = useState(false);
+  const [uploadedDataUrl, setUploadedDataUrl] = useState<string | null>(null);
+  const [uploadedFile, setUploadedFile] = useState<File | null>(null);
+  const [cropDataUrl, setCropDataUrl] = useState<string>(initialImageUrl ?? '');
+
+  const isCircular = viewMode === 'circle';
+
+  const fallbackInputReference = useRef<React.ElementRef<'input'> | null>(null);
+
+  const handleFileRead = useCallback((url: string) => {
+    setUploadedDataUrl(url);
+    setIsSelecting(true);
+  }, []);
+
+  const handleImageSelected = useCallback(
+    ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+      if (!hasSingleFile(target.files)) {
+        return;
+      }
+      const [file] = target.files;
+      setUploadedFile(file);
+      readFile(file, handleFileRead);
+    },
+    [handleFileRead],
+  );
+
+  const handleModalClose = useCallback(() => {
+    setIsSelecting(false);
+  }, []);
+
+  const handleSelectImageClick = useCallback(() => {
+    if (fallbackInputReference.current) {
+      fallbackInputReference.current.click();
+    }
+  }, [fallbackInputReference]);
+
+  const handleCropSuccess = useCallback(
+    (image: CroppedImage | undefined) => {
+      if (uploadedFile && image) {
+        const inputValue = createFileList(
+          image.blob,
+          changeFileExtension(
+            resultOptions?.overrideFilename ?? uploadedFile.name,
+            resultOptions?.format ?? Default.OUTPUT_FORMAT,
+          ),
+        );
+
+        setCropDataUrl(image.link);
+        setValue(name, inputValue as PathValue<T, K>);
+      }
+      setIsSelecting(false);
+    },
+    [name, resultOptions, setValue, uploadedFile],
+  );
+
+  const { ref: setReactHookFormReference, ...inputReactHookFormProperties } =
+    useMemo(
+      () =>
+        register(name, {
+          onChange: handleImageSelected,
+        }),
+      [handleImageSelected, name, register],
+    );
+
+  const getInputReference = useCallback(
+    (instance: HTMLInputElement): void => {
+      if (inputReference) {
+        inputReference.current = instance;
+      }
+      fallbackInputReference.current = instance;
+      setReactHookFormReference(instance);
+    },
+    [inputReference, setReactHookFormReference],
+  );
+
+  return (
+    <div className={getValidClassNames(styles.wrapper, className)}>
+      <input
+        ref={getInputReference}
+        type="file"
+        accept="image/*"
+        hidden
+        {...inputReactHookFormProperties}
+      />
+      {!inputReference && (
+        <Icon
+          iconName="edit"
+          onClick={handleSelectImageClick}
+          size="lg"
+          className={styles.editIcon}
+        />
+      )}
+      <img
+        className={isCircular ? styles.croppedImageCircle : undefined}
```

Comment:
can we assign here some default styles instead of undefined?

Reply:
undefined means do not add anything. What do you mean by saying "default styles"?
This code either add "border-radius: 100%" or not. If isCircular is false, we shouldnt have any additional classes.
I'll write is as
getValidClassNames(isCircular && styles.croppedImageCircle)
if you like it that way

Reply:
if isCircular is false, then how it looks like, which styles apply? 
about using getValidClassNames - it is used for many styles, not in cases when we have only 1 style

Reply:
for the case className={isCircular ? styles.croppedImageCircle : undefined}

isCircular = true:  <img class="croppedImageCircle" ... /> => .croppedImageCircle { border-radius: 100%; }

isCircular = false: <img (nothing here)... /> => no additional styling applied. border radius is default for all images.